source/fab/__main__.py

```diff
@@ -0,0 +1,38 @@
+'''
+Command-line interface to Fab build tool.
+'''
+import argparse
+import logging
+import sys
+
+import fab
+
+
+def parse_cli() -> argparse.Namespace:
+    '''
+    Parse the command line for arguments.
+    '''
+    description='Flexible build system for scientific software.'
+    parser = argparse.ArgumentParser(add_help=False,
+                                     description=description)
+    parser.add_argument('-help', '-h', '--help', action='help',
+                        help='Print this help and exit')
+    parser.add_argument('-version', action='version', version=fab.__version__)
```

Comment:
I think convention is generally to have `--` as a prefix to any flags that aren't single character?  (and on the `-verbose` option below as well)

Reply:
Only on GNU projects. Other realms prefer different approaches. In the particular case of "help" I want to cover all bases I can think of. It should always be easy to get help.

Reply:
Fair enough with help, I agree... I guess we need to decide if we are to follow GNU convention or not (just thinking about a lot of the other tools we use like Rose, Cylc etc. all seem to follow it)

Reply:
In the case of Rose and Cylc, they sort-of follow the GNU standard. The glibc documentation has this to say: https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

As you can see, the GNU "long options" extension specifies `--<name>=<value>` and while Rose and Cylc does support that it also supports `--<name> <value>`.

Sun were proponents of the long option with single hyphen which is reflected in Java. I tend to prefer that mode but it does mean that you can no longer use single letter options. After all does `-cat` mean "cat" or does it mean "c", "a" and "t"?

But there isn't even agreement within Posix tools, look at `find` which offers options like `-name`.

Basically we just have to choose what we like and stick to it.

Reply:
Okay well I think we have one vote for each, @GiorgiaLine for the deciding vote?

Reply:
I think I prefer the GNU convention for this, if only because I like being able to use single letter options.

Reply:
As you wish.